lib/puppet/util/tagging.rb

```diff
@@ -1,7 +1,7 @@
 require 'puppet/util/tag_set'
 
 module Puppet::Util::Tagging
-  ValidTagRegex = /\A[0-9A-Za-z_][0-9A-Za-z_:.-]*\Z/
+  ValidTagRegex = /\A[[:word:]][[[:word:]]:.-]*\Z/
```

Comment:
With @MosesMendoza confirming that PuppetDB code uses RegEx for tag validation as well, I agree that we should not use the Ruby `[[:word:]]` @thallgren and should instead rely on the POSIX compatible `[[:alnum:]]`, which has a Java equivalent in https://docs.oracle.com/javase/6/docs/api/java/util/regex/Pattern.html as `\p{Alnum}`.

/cc @mullr @rbrw